GiEnJul/ClientApp/src/components/InputFields/Validators/InputValidator.tsx

```diff
@@ -97,10 +99,12 @@ const InputValidator: FC<InputValidatorProps> = (
                     break;
             } 
             if (!isValid && viewError) {
-                errorMsg = errorMessages !== undefined ? errorMessages[i] : "dummy";
+                errorMsg = errorMessages !== undefined ? errorMessages[i] : undefined;
+                isAnyError = true;
             }
         }
-        setError(errorMsg);
+        setIsError(isAnyError);
```

Comment:
double setState calls can cause rerender problems, try to avoid it. 
a good way to do this is to create a state that has everything to do with "validating" or "errorState", 
you create an object with the different properties and then setErrorState as needed like so:

if (isValid) {
   setErrorState((prev) => ({ ...prev, hasError: false })) 
} else if (!isValid && errorMessage !== undefined) {
   setErrorState((prev) => ({ 
                    ...prev, hasError: false,
                               errorMessage: errorMsg,
 }))